src/corelibs/U2View/src/ov_msa/find_pattern/FindPatternMsaTask.cpp

```diff
@@ -75,7 +75,7 @@ QList<Task*> FindPatternMsaTask::onSubTaskFinished(Task* subTask) {
 
     if (subTask == searchInSingleSequenceTask) {
         getResultFromTask();
-        if (currentSequenceIndex < settings.msaObj->getNumRows() && totalResultsCounter < settings.findSettings.maxResult2Find) {
+        if (currentSequenceIndex < settings.msaObj->getNumRows() && totalResultsCounter <= settings.findSettings.maxResult2Find) {
```

Comment:
There is no need to create a new search task if there are already settings.findSettings.maxResult2Find results found.